packages/components/menu/src/MenuList/MenuList.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import { CommonProps, PropsWithHTMLElement } from '@contentful/f36-core';
+import { useMenuContext } from '../MenuContext';
+import { Popover } from '@contentful/f36-popover';
+import { cx } from 'emotion';
+import { getMenuListStyles } from './MenuList.styles';
+import { MenuListHeader } from './MenuListHeader';
+import { MenuListFooter } from './MenuListFooter';
+
+interface MenuListInternalProps extends CommonProps {
+  children?: React.ReactNodeArray;
+}
+
+function assertChild(child: any): child is { type: { displayName: string } } {
+  return Boolean(child?.type?.displayName);
+}
+
+export type MenuListProps = PropsWithHTMLElement<MenuListInternalProps, 'div'>;
+
+const _MenuList = (props: MenuListProps, ref: React.Ref<HTMLDivElement>) => {
+  const {
+    children,
+    testId = 'cf-ui-menu-list',
+    className,
+    ...otherProps
+  } = props;
+
+  const { getMenuListProps } = useMenuContext();
+
+  let header: React.ReactElement | null = null;
+  let footer: React.ReactElement | null = null;
+  const items: React.ReactElement[] = [];
+
+  React.Children.forEach(children, (child) => {
+    let appendChild = true;
+    if (assertChild(child)) {
+      if (child.type.displayName === MenuListHeader.displayName) {
+        header = (child as unknown) as React.ReactElement;
+        appendChild = false;
+      } else if (child.type.displayName === MenuListFooter.displayName) {
+        footer = (child as unknown) as React.ReactElement;
+        appendChild = false;
+      }
+    }
+    if (appendChild) {
+      items.push((child as unknown) as React.ReactElement);
+    }
```

Comment:
If I understood this right, we want to check if the child has `displayName`
then if the display name is `ListHeader`, we assign it to the `header` variable
if the display name is `ListFooter`, we assign it to the `footer` variable
otherwise, we just append it to the `items` array

is that correct?


What do you think about this 👇 ?
It would reduce the number of if statements and we would not need the appendChild logic
I'm not sure if I got it right though

```jsx
const hasDisplayName = assertChild(child)
    
if (hasDisplayName && child.type.displayName === MenuListHeader.displayName) {
  header = (child as unknown) as React.ReactElement;
} else if (hasDisplayName && child.type.displayName === MenuListFooter.displayName) {
  footer = (child as unknown) as React.ReactElement;
} else {
  items.push((child as unknown) as React.ReactElement);
}
```

This is not a blocker btw, I've already approved the PR
It's just an idea that came to my mind when reading the solution

Reply:
unfortunately, assert functions don't work this way in Typescript, so I had to come with a way I implemented it.

Reply:
Ohhhhh of course!

the solution looks great anyway 🙌